politeiad/backendv2/tstorebe/plugins/comments/cmds.go

```diff
@@ -400,11 +454,255 @@ func (p *commentsPlugin) commentTimestamps(token []byte, commentIDs []uint32, in
 		}
 	}
 
+	// Cache final timestamps
+	err = p.cacheFinalTimestamps(token, cts)
+	if err != nil {
+		return nil, err
+	}
+
 	return &comments.TimestampsReply{
 		Comments: cts,
 	}, nil
 }
 
+// cacheFinalTimestamps accepts a map of comment timestamps, it collects the
+// final timestamps then stores them in the key-value store.
+func (p *commentsPlugin) cacheFinalTimestamps(token []byte, cts map[uint32]comments.CommentTimestamp) error {
+	finalTimestamps, err := finalCommentTimestamps(cts, token)
+	if err != nil {
+		return err
+	}
+	if len(finalTimestamps) > 0 {
+		// Convert final timestamps to map of digests
+		cacheDigests, keys, err := convertFinalTimestampsToDigests(finalTimestamps)
+		if err != nil {
+			return err
+		}
+		// Delete exisiting digests
+		err = p.tstore.CacheDel(keys)
+		if err != nil {
+			return err
+		}
+		// Store updated digests
+		err = p.tstore.CachePut(cacheDigests, false)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// cachedTimestamps retrieves all comment timestamps associated with the given
+// comment IDs from the key-value store.
+func (p *commentsPlugin) cachedTimestamps(token []byte, commentIDs []uint32) (map[string][]byte, error) {
```

Comment:
Why are you returning the byte slices here instead of decoding the timestamps and returning the structures.  That doesn't make any sense.

Reply:
Oh, good point!

Reply:
my bad, didn't see it.